### PR TITLE
texlive: generate fmtutil.cnf

### LINKED ISSUE #101405: texlive.combined.scheme-minimal: can't find the format file `pdflatex.fmt'
I'm trying to use inkscape's latex extension
texlive.combined.scheme-minimal derivation works, but provides a broken pdflatex that complains about missing 'pdflatex.fmt'

### PR DESCRIPTION
**Pull request by xworld21**
###### Motivation for this change
This is a necessary piece towards fixing #101405: we need an accurate `fmtutil.cnf` to ensure `texlinks` generates links only for the installed formats (see https://github.com/NixOS/nixpkgs/issues/101405#issuecomment-714838611).

Bonus side effect: we can drop `|| true` from the `fmtutil` call in `texlive.combine` (this also works as test that the file is generated correctly).

To fix #101405, one should also remove the symlinks from `texlive.bin.core` but I don't know how to do that.

For good measure, I have added a test that we do not miss any format in `fmtutil.cnf`.

Cc @veprbl 

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [X] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Relase notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 130041` at e40614da run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package marked as broken and skipped:</summary>
  <ul>
    <li>spark</li>
  </ul>
</details>
<details>
  <summary>56 packages skipped due to time constraints:</summary>
  <ul>
    <li>adapta-gtk-theme</li>
    <li>asymptote</li>
    <li>autofs5</li>
    <li>bibletime</li>
    <li>ccache</li>
    <li>ccacheStdenv</li>
    <li>ccacheWrapper</li>
    <li>clevis</li>
    <li>denemo</li>
    <li>digikam</li>
    <li>...</li>
  </ul>
</details>
<details>
  <summary>56 packages built successfully:</summary>
  <ul>
    <li>R</li>
    <li>apostrophe</li>
    <li>asciidoc-full</li>
    <li>asciidoc-full-with-plugins</li>
    <li>auctex</li>
    <li>btrbk</li>
    <li>cddlib</li>
    <li>chrome-gnome-shell</li>
    <li>dblatex</li>
    <li>dblatexFull</li>
    <li>diffoscope</li>
    <li>disorderfs</li>
    <li>dot2tex (python38Packages.dot2tex)</li>
    <li>dwarf-therapist (dwarf-fortress-packages.dwarf-therapist)</li>
    <li>dwarf-fortress-packages.dwarf-therapist-original</li>
    <li>frescobaldi</li>
    <li>gnome.gnome-session (gnome.gnome_session)</li>
    <li>gnome.gnome-shell (gnome.gnome_shell)</li>
    <li>gnome.gnome-tweaks (gnome.gnome-tweak-tool)</li>
    <li>gnomeExtensions.easyScreenCast</li>
    <li>gnomeExtensions.gsconnect</li>
    <li>gnomeExtensions.night-theme-switcher</li>
    <li>gp2c</li>
    <li>rep (kakounePlugins.rep)</li>
    <li>lcalc</li>
    <li>lilypond</li>
    <li>luksmeta</li>
    <li>manim</li>
    <li>modem-manager-gui</li>
    <li>ne</li>
    <li>nuweb</li>
    <li>paperwork</li>
    <li>pari</li>
    <li>perl532Packages.Po4a</li>
    <li>postgresqlPackages.plr (postgresql11Packages.plr)</li>
    <li>python38Packages.cnvkit</li>
    <li>python38Packages.cysignals</li>
    <li>python38Packages.pypandoc</li>
    <li>python38Packages.sphinxcontrib-bayesnet</li>
    <li>python38Packages.sphinxcontrib-tikz</li>
    <li>python39Packages.cysignals</li>
    <li>python39Packages.dot2tex</li>
    <li>python39Packages.pypandoc</li>
    <li>python39Packages.sphinxcontrib-bayesnet</li>
    <li>python39Packages.sphinxcontrib-tikz</li>
    <li>solfege</li>
    <li>tang</li>
    <li>texlive.combined.scheme-basic</li>
    <li>texlive.combined.scheme-context</li>
    <li>texlive.combined.scheme-full</li>
    <li>texlive.combined.scheme-infraonly</li>
    <li>texlive.combined.scheme-medium</li>
    <li>texlive.combined.scheme-minimal</li>
    <li>texlive.combined.scheme-small</li>
    <li>udiskie</li>
    <li>zettlr</li>
  </ul>
</details>

File: pkgs/tools/typesetting/tex/texlive/combine.nix [hunk 1]
```diff
@@ -212,7 +224,7 @@ in (buildEnv {
 
     perl `type -P mktexlsr.pl` ./share/texmf
     ${bin.texlinks} "$out/bin" && wrapBin
-    (perl `type -P fmtutil.pl` --sys --all || true) | grep '^fmtutil' # too verbose
+    perl `type -P fmtutil.pl` --sys --all | grep '^fmtutil' # too verbose
```
Comment (veprbl) [inline pkgs/tools/typesetting/tex/texlive/combine.nix:227]: Little known fact: `set -e` doesn't catch things like `false | cat`

**Comment by veprbl** [comment]
@GrahamcOfBorg build tests.texlive

File: pkgs/tools/typesetting/tex/texlive/combine.nix (hunk 1 above)
Comment (xworld21) [reply pkgs/tools/typesetting/tex/texlive/combine.nix:227]: Uhm, the intention is for the build to fail if `fmtutil` fails. At the moment, the build does fail if e.g. `fmtutil.cnf` contains formats for which the binaries are not available.

**Pull request merged**